app/client/jsx/ModalFactory.jsx

```diff
@@ -0,0 +1,29 @@
+import _ from 'lodash'
+import React from 'react'
+import Map from './Map.jsx'
+import EventList from './EventList.jsx'
+import EmailForm from './EmailForm.jsx'
+import CustomComponents from './CustomComponents'
+
+export default ({ room, modal, props = {} }) => {
+    if (modal in CustomComponents) {
+        const Component = CustomComponents[modal]
+        return <Component room={room} {...props} />
+    }
+    const map = (
+        <Map
+            onRoomClick={room.onSwitchRoom.bind(room)}
+            handleCloseMap={() => room.handleModalChange(null)}>
+        </Map>
+    )
+    const events = (
+        <EventList
+            rooms={room.props.rooms}
+            events={room.props.events}>
+        </EventList>
+    )
+    const email = (
+        <EmailForm user={room.props.user}></EmailForm>
+    )
+    return { map, events, email }[modal] || null
```

Comment:
probably don't want to create components that get thrown away, in case we end up writing one that has side effects

Reply:
What do you mean?

Reply:
I'm going to merge this so I can continue working on TST customization, but PM me!

Reply:
this piece of code appears to instantiate all three of the map, events, and email components, then discard them if none of them is the component that is asked for.

If any of these components makes network calls, manipulates some kind of state, etc, we might end up with unexpected behavior. At the very least this is kind of wasteful